src/MicrosoftTeams.ts

```diff
@@ -417,25 +441,132 @@ namespace microsoftTeams
         {
             ensureInitialized(frameContexts.content, frameContexts.settings, frameContexts.remove);
 
+            // Open an authentication window with the parameters provided by the caller
+            openAuthenticationWindow(authenticateParameters.url, authenticateParameters.width, authenticateParameters.height);
+
+            authParams = authenticateParameters;
+        }
+
+        function closeAuthenticationWindow(): void
+        {
+            // Stop monitoring the authentication window
+            stopAuthenticationWindowMonitor();
+
+            // Try to close the authentication window and clear all properties associated with it
+            try
+            {
+                if (childWindow)
+                {
+                    childWindow.close();
+                }
+            }
+            finally
+            {
+                childWindow = null;
+                childOrigin = null;
+            }
+        }
+
+        function openAuthenticationWindow(url: string, width?: number, height?: number): void
```

Comment:
>width?: number, height?: number [](start = 55, length = 31)

These two params actually aren't optional.

Note that there's a difference b/t optional params (not provided in the invocation) and passing null/undefined as the value.